src/veritysetup/veritysetup.c

```diff
@@ -178,6 +182,21 @@ static int parse_options(const char *options) {
                         r = free_and_strdup(&arg_hash, val);
                         if (r < 0)
                                 return log_oom();
+                } else if ((val = startswith(word, "fec-device="))) {
+
+                        r = free_and_strdup(&arg_fec_what, val);
```

Comment:
validate with path_is_absolute() and path_is_normalized()?

Reply:
Well, I guess I should also translate the `UUID=` into a `/dev/disk/by-uuid` path.

Reply:
there's a helper for that already: fstab_node_to_udev_node()

Reply:
Well, I have tested with files instead of block devices, with relative path. Do you still want to check for `path_is_absolute()`?

However, I do not handle case where `fec-device=` is `PARTUUID=` (`UUID=`, ...) device path; nor `argv[3]` (aka `data_device`) nor `argv[4]` (aka `hash_device`).

I guess it is the duty of the generator, isn't it? :/

Reply:
a relative path would be relative to what? the root fs? /etc/ (because /etc/veritytyab is located there)? I am pretty sure we should just disallow relative path names.

Generally our approach is to parse+validated whenever we accept user data, and then use internally only parsed/validated data, and format to string again on output.

Thus, we should really use fstab_node_to_udev_node() here, and also do the check for path_is_absolute() and path_is_normalized() here afterwards.

Reply:
done.

I hope I did it right.